src/containerbuddy/ips_test.go

```diff
@@ -0,0 +1,406 @@
+package main
+
+import (
+	"encoding/json"
+	"math/rand"
+	"net"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+)
+
+// ------------------------------------------
+// Test setup with mock services
+
+type MockAddr struct {
+	NetworkAttr string
+	StringAttr  string
+}
+
+func (addr MockAddr) Network() string {
+	return addr.NetworkAttr
+}
+
+func (addr MockAddr) String() string {
+	return addr.StringAttr
+}
+
+func TestParseInterfaces(t *testing.T) {
+	if interfaces, err := parseInterfaces(nil); err != nil {
+		t.Errorf("Unexpected parse error: %s", err.Error())
+	} else if len(interfaces) > 0 {
+		t.Errorf("Expected no interfaces, but got %s", interfaces)
+	}
+
+	json1 := json.RawMessage(`"eth0"`)
+	expected1 := []string{"eth0"}
+	if interfaces, err := parseInterfaces(json1); err != nil {
+		t.Errorf("Unexpected parse error: %s", err.Error())
+	} else if !reflect.DeepEqual(interfaces, expected1) {
+		t.Errorf("Expected %s, got: %s", expected1, interfaces)
+	}
+
+	json2 := json.RawMessage(`["ethwe","eth0"]`)
+	expected2 := []string{"ethwe", "eth0"}
+	if interfaces, err := parseInterfaces(json2); err != nil {
+		t.Errorf("Unexpected parse error: %s", err.Error())
+	} else if !reflect.DeepEqual(interfaces, expected2) {
+		t.Errorf("Expected %s, got: %s", expected2, interfaces)
+	}
+
+	json3 := json.RawMessage(`{ "a": true }`)
+	if _, err := parseInterfaces(json3); err == nil {
+		t.Errorf("Expected parse error for json3")
+	}
+}
+
+func TestGetIp(t *testing.T) {
+	if ip, _ := GetIP([]string{}); ip == "" {
+		t.Errorf("Expected default interface to yield an IP, but got nothing.")
+	}
+	if ip, _ := GetIP(nil); ip == "" {
+		t.Errorf("Expected default interface to yield an IP, but got nothing.")
+	}
+	if ip, _ := GetIP([]string{"eth0"}); ip == "" {
+		t.Errorf("Expected to find IP for eth0, but found nothing.")
+	}
+	if ip, _ := GetIP([]string{"eth0", "lo"}); ip == "127.0.0.1" {
+		t.Errorf("Expected to find eth0 ip, but found loopback instead")
+	}
+	if ip, _ := GetIP([]string{"lo", "eth0"}); ip != "127.0.0.1" {
+		t.Errorf("Expected to find loopback ip, but found: %s", ip)
+	}
+	if ip, err := GetIP([]string{"interface-does-not-exist"}); err == nil {
+		t.Errorf("Expected interface not found, but instead got an IP: %s", ip)
+	}
+}
+
+func TestOnChangeCmd(t *testing.T) {
```

Comment:
I think this `TestOnChangeCmd` and `TestHealthCheck` got moved to this file accidentally.